**1. INTRODUCTION TO OUR DATASET.**

COVID-19 is the disease caused by a coronavirus called SARS-CoV-2 (Severe acute respiratory syndrome coronavirus 2).  It was first discovered on 31st December 2019, following a report of a cluster of cases of so-called viral pneumonia in Wuhan, People’s Republic of China.
Since its identification, this virus has impacted the world in all sectors, it's one of the hardest periods humanbeings have encountered.

The most common symptoms of COVID-19 are fever, chills,sore throat etc.

It is difficult to know how many people have COVID-19, so we're going to analyze this dataset in other to gain deeper insight into COVID-19.

**ACCESSING OUR DATASET.**

In [65]:
#Importing our modules
import pandas as pd
import plotly.express as px
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.graph_objs as go



daily_df = pd.read_csv("/content/worldometer_coronavirus_daily_data.csv")
summary_df = pd.read_csv("/content/worldometer_coronavirus_summary_data.csv")

**2. PRELIMINARY DATA EXPLORATION.**

In [66]:
daily_df.shape

(184787, 7)

In [67]:
summary_df.shape

(226, 12)

There's 184784 Rows and 7 Columns in daily dataset, while the summary dataset contains 226 Rows and 12 Columns.

In [68]:
daily_df.columns

Index(['date', 'country', 'cumulative_total_cases', 'daily_new_cases',
       'active_cases', 'cumulative_total_deaths', 'daily_new_deaths'],
      dtype='object')

In [69]:
summary_df.columns

Index(['country', 'continent', 'total_confirmed', 'total_deaths',
       'total_recovered', 'active_cases', 'serious_or_critical',
       'total_cases_per_1m_population', 'total_deaths_per_1m_population',
       'total_tests', 'total_tests_per_1m_population', 'population'],
      dtype='object')

In [70]:
daily_df.head(5)

,date,country,cumulative_total_cases,daily_new_cases,active_cases,cumulative_total_deaths,daily_new_deaths
0,2020-2-15,Afghanistan,0.0,NaN,0.0,0.0,NaN
1,2020-2-16,Afghanistan,0.0,NaN,0.0,0.0,NaN
2,2020-2-17,Afghanistan,0.0,NaN,0.0,0.0,NaN
3,2020-2-18,Afghanistan,0.0,NaN,0.0,0.0,NaN
4,2020-2-19,Afghanistan,0.0,NaN,0.0,0.0,NaN


In [71]:
daily_df.tail(10)

,date,country,cumulative_total_cases,daily_new_cases,active_cases,cumulative_total_deaths,daily_new_deaths
184777,2022-5-05,Zimbabwe,248150.0,100.0,739.0,5472.0,1.0
184778,2022-5-06,Zimbabwe,248214.0,64.0,765.0,5473.0,1.0
184779,2022-5-07,Zimbabwe,248269.0,55.0,785.0,5476.0,3.0
184780,2022-5-08,Zimbabwe,248352.0,83.0,791.0,5476.0,0.0
184781,2022-5-09,Zimbabwe,248536.0,184.0,939.0,5479.0,3.0
184782,2022-5-10,Zimbabwe,248642.0,106.0,963.0,5481.0,2.0
184783,2022-5-11,Zimbabwe,248778.0,136.0,1039.0,5481.0,0.0
184784,2022-5-12,Zimbabwe,248943.0,165.0,1158.0,5481.0,0.0
184785,2022-5-13,Zimbabwe,249131.0,188.0,1283.0,5482.0,1.0
184786,2022-5-14,Zimbabwe,249206.0,75.0,1307.0,5482.0,0.0


In [72]:
summary_df.head(10)

,country,continent,total_confirmed,total_deaths,total_recovered,active_cases,serious_or_critical,total_cases_per_1m_population,total_deaths_per_1m_population,total_tests,total_tests_per_1m_population,population
0,Afghanistan,Asia,179267,7690.0,162202.0,9375.0,1124.0,4420,190.0,951337.0,23455.0,40560636
1,Albania,Europe,275574,3497.0,271826.0,251.0,2.0,95954,1218.0,1817530.0,632857.0,2871945
2,Algeria,Africa,265816,6875.0,178371.0,80570.0,6.0,5865,152.0,230861.0,5093.0,45325517
3,Andorra,Europe,42156,153.0,41021.0,982.0,14.0,543983,1974.0,249838.0,3223924.0,77495
4,Angola,Africa,99194,1900.0,97149.0,145.0,NaN,2853,55.0,1499795.0,43136.0,34769277
5,Anguilla,North America,2984,9.0,2916.0,59.0,4.0,195646,590.0,51382.0,3368870.0,15252
6,Antigua And Barbuda,North America,7721,137.0,7511.0,73.0,1.0,77646,1378.0,18901.0,190076.0,99439
7,Argentina,South America,9101319,128729.0,8895999.0,76591.0,372.0,197992,2800.0,35716069.0,776974.0,45968174
8,Armenia,Asia,422896,8623.0,412048.0,2225.0,NaN,142219,2900.0,3068217.0,1031834.0,2973558
9,Aruba,North America,35693,213.0,35199.0,281.0,NaN,331689,1979.0,177885.0,1653053.0,107610


In [73]:
summary_df.tail(10)

,country,continent,total_confirmed,total_deaths,total_recovered,active_cases,serious_or_critical,total_cases_per_1m_population,total_deaths_per_1m_population,total_tests,total_tests_per_1m_population,population
216,USA,North America,84209473,1026646.0,81244260.0,1938567.0,1941.0,251659,3068.0,1.016883e+09,3038939.0,334617623
217,Uzbekistan,Asia,238802,1637.0,236974.0,191.0,23.0,6947,48.0,1.377915e+06,40088.0,34372515
218,Vanuatu,Australia/Oceania,8457,14.0,7974.0,469.0,NaN,26386,44.0,2.497600e+04,77926.0,320508
219,Venezuela,South America,522921,5711.0,516170.0,1040.0,230.0,18487,202.0,3.359014e+06,118752.0,28285909
220,Viet Nam,Asia,10696630,43065.0,9355040.0,1298525.0,340.0,108080,435.0,8.581148e+07,867048.0,98969721
221,Wallis And Futuna Islands,Australia/Oceania,454,7.0,438.0,9.0,NaN,41755,644.0,2.050800e+04,1886140.0,10873
222,Western Sahara,Africa,10,1.0,9.0,0.0,NaN,16,2.0,NaN,NaN,624681
223,Yemen,Asia,11819,2149.0,9009.0,661.0,23.0,381,69.0,2.652530e+05,8543.0,31049015
224,Zambia,Africa,320591,3983.0,315997.0,611.0,NaN,16575,206.0,3.452554e+06,178497.0,19342381
225,Zimbabwe,Africa,249206,5482.0,242417.0,1307.0,12.0,16324,359.0,2.287793e+06,149863.0,15265849


In [74]:
print(f' Is there any duplicate in the Daily DataFrame? {daily_df.duplicated().values.any()}.')

 Is there any duplicate in the Daily DataFrame? False.


In [75]:
print(f' Is there any duplicate in the Summary DataFrame? {summary_df.duplicated().values.any()}.')

 Is there any duplicate in the Summary DataFrame? False.


In [76]:
daily_df.describe()

,cumulative_total_cases,daily_new_cases,active_cases,cumulative_total_deaths,daily_new_deaths
count,1.847870e+05,174329.000000,1.667470e+05,1.782270e+05,157850.000000
mean,7.251089e+05,2987.633285,6.239283e+04,1.388600e+04,39.831834
std,3.681471e+06,17803.232663,3.955641e+05,6.049521e+04,181.102770
min,0.000000e+00,-322.000000,-1.432100e+04,0.000000e+00,-39.000000
25%,1.099000e+03,0.000000,6.000000e+01,2.400000e+01,0.000000
50%,1.775600e+04,58.000000,1.386000e+03,3.040000e+02,1.000000
75%,2.238085e+05,728.000000,1.462050e+04,4.111000e+03,12.000000
max,8.420947e+07,909610.000000,1.793543e+07,1.026646e+06,5093.000000


From the table above we can see that there is 7.25e+05 (7.25 * 100,000 = 725,000) cumulative total cases, 2988 daily new cases, 62400 active cases, 13900 cumulative total deaths and 40 daily new deaths.

In [77]:
summary_df.describe()

,total_confirmed,total_deaths,total_recovered,active_cases,serious_or_critical,total_cases_per_1m_population,total_deaths_per_1m_population,total_tests,total_tests_per_1m_population,population
count,2.260000e+02,2.180000e+02,2.040000e+02,2.040000e+02,145.000000,226.000000,218.000000,2.120000e+02,2.120000e+02,2.260000e+02
mean,2.305651e+06,2.884442e+04,2.256851e+06,6.861029e+04,269.482759,148156.809735,1157.550459,2.987445e+07,1.944650e+06,3.495521e+07
std,7.575510e+06,9.971254e+04,7.613357e+06,2.350430e+05,883.174727,155202.909225,1212.833590,1.079635e+08,3.318400e+06,1.390338e+08
min,2.000000e+00,1.000000e+00,1.000000e+00,0.000000e+00,1.000000,16.000000,2.000000,5.117000e+03,5.093000e+03,8.050000e+02
25%,2.412600e+04,2.370000e+02,1.619300e+04,2.397500e+02,5.000000,11748.250000,172.500000,3.472968e+05,1.667260e+05,5.605125e+05
50%,1.793750e+05,2.251500e+03,1.372740e+05,1.634000e+03,17.000000,98271.500000,768.000000,2.238918e+06,7.753355e+05,5.800570e+06
75%,1.090902e+06,1.400650e+04,1.006245e+06,1.929675e+04,139.000000,255632.750000,1850.000000,1.233862e+07,2.267408e+06,2.187284e+07
max,8.420947e+07,1.026646e+06,8.124426e+07,1.938567e+06,8318.000000,704302.000000,6297.000000,1.016883e+09,2.184247e+07,1.439324e+09


From the table above we can see that there is 2310000 total confirmed cases, 28800 total deaths, 2260000 total recovered, 68600 active cases, 269 serious cases, 148156 total cases per one million population, 1158 total deaths per one million population, 29900000 total tests,1940000 total tests per one million population and total population of 35,000,000.

**3. CASES BY COUNTRY.**

In [78]:
#Cases by country
cases_by_country = daily_df.groupby('country')[[
    'cumulative_total_cases',
    'daily_new_cases',
    'active_cases',
    'cumulative_total_deaths',
    'daily_new_deaths'
]].sum()
cases_by_country.head(10)

,cumulative_total_cases,daily_new_cases,active_cases,cumulative_total_deaths,daily_new_deaths
country,,,,,
Afghanistan,7.115616e+07,179242.0,9903666.0,3094865.0,7690.0
Albania,9.011648e+07,275572.0,7412581.0,1418680.0,3496.0
Algeria,1.021373e+08,265815.0,29762940.0,2844496.0,6875.0
Andorra,1.075871e+07,42155.0,509375.0,80172.0,153.0
Angola,2.920640e+07,99193.0,2264022.0,658512.0,1898.0
Anguilla,4.673670e+05,2982.0,15776.0,1368.0,8.0
Antigua And Barbuda,1.648447e+06,7720.0,112719.0,37159.0,136.0
Argentina,2.730257e+09,9101318.0,116720051.0,52798218.0,128728.0
Armenia,1.546468e+08,422895.0,8443240.0,3110491.0,8623.0


**3.1 Country with the maximum cases.**

In [79]:
#Country with the maximum cases
cases_by_country.idxmax()

,0
cumulative_total_cases,USA
daily_new_cases,USA
active_cases,USA
cumulative_total_deaths,USA
daily_new_deaths,USA


From the table above, we see that USA has the maximum cumulative total cases, active cases, daily new cases, cumulative total deaths and daily new deaths.


**3.2 Country with the minimum case.**

In [80]:
#Country with the minimum cases
cases_by_country.idxmin()

,0
cumulative_total_cases,Nauru
daily_new_cases,Saint Helena
active_cases,Belarus
cumulative_total_deaths,China Macao Sar
daily_new_deaths,China Macao Sar


From the table above, we can see that a country can have the minimum cumulative total cases, even though they don't have the minimum daily new or active cases.

In [81]:
summary_df.head()

,country,continent,total_confirmed,total_deaths,total_recovered,active_cases,serious_or_critical,total_cases_per_1m_population,total_deaths_per_1m_population,total_tests,total_tests_per_1m_population,population
0,Afghanistan,Asia,179267,7690.0,162202.0,9375.0,1124.0,4420,190.0,951337.0,23455.0,40560636
1,Albania,Europe,275574,3497.0,271826.0,251.0,2.0,95954,1218.0,1817530.0,632857.0,2871945
2,Algeria,Africa,265816,6875.0,178371.0,80570.0,6.0,5865,152.0,230861.0,5093.0,45325517
3,Andorra,Europe,42156,153.0,41021.0,982.0,14.0,543983,1974.0,249838.0,3223924.0,77495
4,Angola,Africa,99194,1900.0,97149.0,145.0,NaN,2853,55.0,1499795.0,43136.0,34769277


**3.3 Mortality Rate.**

Mortality rate is the measure of the number of deaths to the total population.

In [82]:
#Creating a column called mortality_rate
summary_df["mortality_rate"] = summary_df["total_deaths"] / summary_df["population"] * 100
summary_df.head()

,country,continent,total_confirmed,total_deaths,total_recovered,active_cases,serious_or_critical,total_cases_per_1m_population,total_deaths_per_1m_population,total_tests,total_tests_per_1m_population,population,mortality_rate
0,Afghanistan,Asia,179267,7690.0,162202.0,9375.0,1124.0,4420,190.0,951337.0,23455.0,40560636,0.018959
1,Albania,Europe,275574,3497.0,271826.0,251.0,2.0,95954,1218.0,1817530.0,632857.0,2871945,0.121764
2,Algeria,Africa,265816,6875.0,178371.0,80570.0,6.0,5865,152.0,230861.0,5093.0,45325517,0.015168
3,Andorra,Europe,42156,153.0,41021.0,982.0,14.0,543983,1974.0,249838.0,3223924.0,77495,0.197432
4,Angola,Africa,99194,1900.0,97149.0,145.0,NaN,2853,55.0,1499795.0,43136.0,34769277,0.005465


**3.4 Fatality Rate.**

Fatality Rate is the measure of deaths caused by a certain disease or thing, compared to the tottal number of people affected by the disease or thing

In [83]:
summary_df["fatality_rate"] = summary_df["total_deaths"] / summary_df["total_confirmed"] * 100

summary_df.head()

,country,continent,total_confirmed,total_deaths,total_recovered,active_cases,serious_or_critical,total_cases_per_1m_population,total_deaths_per_1m_population,total_tests,total_tests_per_1m_population,population,mortality_rate,fatality_rate
0,Afghanistan,Asia,179267,7690.0,162202.0,9375.0,1124.0,4420,190.0,951337.0,23455.0,40560636,0.018959,4.289691
1,Albania,Europe,275574,3497.0,271826.0,251.0,2.0,95954,1218.0,1817530.0,632857.0,2871945,0.121764,1.268988
2,Algeria,Africa,265816,6875.0,178371.0,80570.0,6.0,5865,152.0,230861.0,5093.0,45325517,0.015168,2.586376
3,Andorra,Europe,42156,153.0,41021.0,982.0,14.0,543983,1974.0,249838.0,3223924.0,77495,0.197432,0.362938
4,Angola,Africa,99194,1900.0,97149.0,145.0,NaN,2853,55.0,1499795.0,43136.0,34769277,0.005465,1.915438


In [84]:
#Cases by continent
cases_by_continent = summary_df.groupby('continent')[[
    'total_confirmed',
    'total_deaths',
    'total_recovered',
    'active_cases',
    'serious_or_critical',
    'population',
    'mortality_rate',
    'fatality_rate'
]].sum()
cases_by_continent.head(10)

,total_confirmed,total_deaths,total_recovered,active_cases,serious_or_critical,population,mortality_rate,fatality_rate
continent,,,,,,,,
Africa,12042400,254319.0,10137200.0,497766.0,966.0,1400358940,1.797925,115.060492
Asia,149999659,1427939.0,126145273.0,3260318.0,11768.0,4672455975,3.142831,72.113784
Australia/Oceania,7942867,11413.0,7403813.0,455469.0,162.0,43505620,0.646560,7.807664
Europe,194330079,1830655.0,170861871.0,5841832.0,8050.0,748470690,11.333471,47.078745
North America,99625662,1467234.0,94818163.0,3329149.0,7460.0,597707388,5.158086,46.685297
South America,57136485,1296523.0,51031313.0,611966.0,10669.0,437379735,3.155621,28.821005


**VISUALIZATION**

**1. Mortality Rate by Countries**

In [85]:
line = px.line(
    summary_df,
    x = 'country',
    y = 'mortality_rate',
    color = 'continent',
    title = 'Mortality Rate by Countries'
)
line.show()

**2. Fatality Rate by Countries.**

In [86]:
line = px.line(
    summary_df,
    x = 'country',
    y = 'fatality_rate',
    color = 'continent',
    title = 'Fatality Rate by Countries'
)
line.show()

**3. Total Confirmed Cases by Countries**

In [87]:
line = px.line(
    summary_df,
    x = 'country',
    y = 'total_confirmed',
    color = 'continent',
    title = 'Total Confirmed Cases by Countries'
)
line.show()

**4. Total Deaths by Countries.**

In [88]:
line = px.line(
    summary_df,
    x = 'country',
    y = 'total_deaths',
    color = 'continent',
    title = 'Total Deaths by Countries'
)
line.show()

**5. CHOROPLETH**

**5.1 Coronavirus Confirmed Cases by Countries.**

In [89]:
def choropleth(Stats, ColorbarTitle, GraphTitle):
    fig = go.Figure(
        data = go.Choropleth(
            locations = summary_df['country'],
            locationmode = 'country names',
            z = Stats,
            text = summary_df['country'],
            colorscale = 'reds',
            autocolorscale = False,
            reversescale = False,
            marker_line_color = 'darkgray',
            marker_line_width = 0.5,
            colorbar_title = ColorbarTitle,
        )
    )

    fig.update_layout(
        title_text = GraphTitle,
        geo = dict(
            showframe = False,
            showcoastlines = True,
            projection_type = 'equirectangular'
        ),
    )

    fig.show()


In [90]:
choropleth(
    summary_df["total_confirmed"],
    "Confirmed Cases",
    "Coronavirus Confirmed Cases by Countries"
)

**5.2 Coronavirus Total Deaths by Countries.**

In [91]:
choropleth(
    summary_df["total_deaths"],
    "Deaths",
    "Coronavirus Total Death by Countries"
)

**5.3 Coronavirus Recovered Cases by Countries.**

In [92]:
choropleth(
    summary_df["total_recovered"],
    "Recovered Cases",
    "Coronavirus Recovered Cases by Countries"
)

**5.4 Coronavirus Active Cases by Countries.**

In [93]:
choropleth(
    summary_df["active_cases"],
    "Active Cases",
    "Coronavirus Active Cases"
)

**6. CORONAVIRUS CASES BY CONTINENTS.**

In [94]:
cases_by_continents = summary_df.groupby(
    "continent",
    as_index = False
)[[
    "total_confirmed",
    "total_deaths",
    "total_recovered",
    "active_cases",
    "serious_or_critical"
]].sum()

cases_by_continents.head()

,continent,total_confirmed,total_deaths,total_recovered,active_cases,serious_or_critical
0,Africa,12042400,254319.0,10137200.0,497766.0,966.0
1,Asia,149999659,1427939.0,126145273.0,3260318.0,11768.0
2,Australia/Oceania,7942867,11413.0,7403813.0,455469.0,162.0
3,Europe,194330079,1830655.0,170861871.0,5841832.0,8050.0
4,North America,99625662,1467234.0,94818163.0,3329149.0,7460.0


In [95]:
africa = cases_by_continents[
    cases_by_continents["continent"] == "Africa"
]

asia = cases_by_continents[
    cases_by_continents["continent"] == "Asia"
]

north_america = cases_by_continents[
    cases_by_continents["continent"] == "North America"
]

south_america = cases_by_continents[
    cases_by_continents["continent"] == "South America"
]

europe = cases_by_continents[
    cases_by_continents["continent"] == "Europe"
]

In [96]:
def pie_chart(continent, text):
    trace = go.Pie(
        labels = [
            'Total Confirmed',
            'Total Deaths',
            'Total Recovered',
            'Total Active',
            'Critical Cases'
        ],
        values = [
            continent.total_confirmed.sum(),
            continent.total_deaths.sum(),
            continent.total_recovered.sum(),
            continent.active_cases.sum(),
            continent.serious_or_critical.sum()
        ],
        title_font_size = 20,
        hovertemplate = "<b>%{label}</b><b><br>%{value}<br><i>%{percent}</i>",
        textinfo = 'percent',
        textposition = 'inside',
        showlegend = True,
        marker = dict(
            colors = [
                "#ff5148",
                "#000201",
                "#61f7ff",
                "#ffb861",
                "#61ffb8"
            ],
            line = dict(
                color = '#000000',
                width = 2
            )
        )
    )
    fig = go.Figure(data=[trace])
    fig.update_layout(title_text = text)
    fig.show()

**6.1 Coronavirus Cases in Africa.**

In [97]:
#Cases in Africa
pie_chart(africa, "Coronavirus Cases in Africa")

**6.2 Coronavirus Cases in Asia.**

In [98]:
#Cases in Asia
pie_chart(asia, "Coronavirus Cases in Asia")

**6.3 Coronavirus Cases in North America.**

In [99]:
pie_chart(north_america, "Coronavirus Cases in North America")

**6.4 Coronavirus Cases in South America.**

In [100]:
pie_chart(south_america, "Coronavirus Cases in South America")

**6.5 Coronavirus Cases in Europe.**

In [101]:
pie_chart(europe, "Coronavirus Cases in Europe")

**7. TOP AFFECTED COUNTRIES.**

In [102]:
top = summary_df.sort_values(
    by = "total_deaths",
    ascending = False
)[:5]

top

,country,continent,total_confirmed,total_deaths,total_recovered,active_cases,serious_or_critical,total_cases_per_1m_population,total_deaths_per_1m_population,total_tests,total_tests_per_1m_population,population,mortality_rate,fatality_rate
216,USA,North America,84209473,1026646.0,81244260.0,1938567.0,1941.0,251659,3068.0,1.016883e+09,3038939.0,334617623,0.306812,1.219157
26,Brazil,South America,30682094,664920.0,29718402.0,298772.0,8318.0,142460,3087.0,6.377617e+07,296119.0,215373503,0.308729,2.167127
94,India,Asia,43121599,524214.0,42579693.0,17692.0,698.0,30686,373.0,8.438369e+08,600479.0,1405273033,0.037303,1.215665
165,Russia,Europe,18260293,377670.0,17647179.0,235444.0,2300.0,125027,2586.0,2.734000e+08,1871949.0,146050996,0.258588,2.068258
131,Mexico,North America,5745652,324465.0,5044893.0,376294.0,4798.0,43708,2468.0,1.585549e+07,120615.0,131455607,0.246825,5.647140


**8. PERCENTAGE OF CONTINENTS IN DATASET.**

In [103]:
by_continents = summary_df.continent.value_counts()
by_continents

,count
continent,
Africa,58
Asia,49
Europe,48
North America,39
Australia/Oceania,18
South America,14


In [104]:
fig = px.pie(
    labels= by_continents.index,
    values=by_continents.values,
    title = "Continents",
    names=by_continents.index
)
fig.update_traces(
   textposition='outside', textinfo='percent + label'
)
fig.show()

In [105]:
bar = px.bar(
    x = by_continents.index,
    y = by_continents.values
)
bar.show()

In [106]:
#country_group = summary_df.groupby(["country","continent"], as_index = False).agg({'total_confirmed': pd.Series.count})

#country_group.head(10)

**9. TOP 10 COUNTRY REPORTS.**

In [107]:
def topbar(Yaxis):
    country_cumulative = daily_df.groupby(
        'country'
    )[[
        'cumulative_total_cases',
        'daily_new_cases',
        'active_cases',
        'cumulative_total_deaths',
        'daily_new_deaths'
    ]].sum()
    country_cumulative.sort_values(
        by = Yaxis,
        ascending = False,
        inplace = True
    )
    country_cumulative = country_cumulative.head(10)
    fig = px.bar(
        country_cumulative,
        x = country_cumulative.index,
        y = Yaxis,
        color = country_cumulative.index,
        title = 'Top 10 Countries with Highest ' +Yaxis
    )
    return fig.update_layout(
        title_x = 0.5
    )

**9.1 Top 10 Countries With Highest Cumulative Total Cases.**

In [108]:
topbar('cumulative_total_cases')

**9.2 Top 10 Countries With Highest Daily New Cases.**

In [109]:
topbar('daily_new_cases')

**9.3 Top 10 Countries With Highest Active Cases.**

In [110]:
topbar('active_cases')

**9.4 Top 10 Countries With Highest Cumulative Total Deaths.**

In [111]:
topbar('cumulative_total_deaths')

**9.5 Top 10 Countries With Highest Daily New Deaths.**

In [112]:
topbar('daily_new_deaths')

**10. COUNTRIES AND THEIR CONFIRMED CASES.**

In [113]:
def casesbar(Yaxis):
    cases_continents = summary_df.groupby('country')[[
        'total_confirmed',
        'total_deaths',
        'total_recovered',
        'serious_or_critical',
        'total_cases_per_1m_population',
        'total_deaths_per_1m_population',
        'total_tests',
        'total_tests_per_1m_population'
    ]].mean()
    cases_continents.sort_values(
        by = Yaxis,
        ascending = False,
        inplace = True
    )
    cases_continents = cases_continents[
        cases_continents[Yaxis] > cases_continents[Yaxis].mean()
    ]
    fig = px.bar(
        cases_continents,
        x = cases_continents.index,
        y = Yaxis,
        color = cases_continents.index,
        title = 'Top Countries with Highest ' +Yaxis
    )
    return fig.update_layout(title_x = 0.5)

**10.1 Top Countries With Highest Total Confirmed.**

In [114]:
casesbar('total_confirmed')

**10.2 Top Countries With Highest Total Deaths.**

In [115]:
casesbar('total_deaths')

**10.3 Top Countries With Highest Total Recovered.**

In [116]:
casesbar('total_recovered')

**10.4 Top Countries With Highest Serious or Critical Cases.**

In [117]:
casesbar('serious_or_critical')

**10.5 Top Countries With Highest Total Cases per 1m Population.**

In [118]:
casesbar('total_cases_per_1m_population')

**10.6 Top Countries With Highest Total Deaths per 1m Population.**

In [119]:
casesbar('total_deaths_per_1m_population')

**10.7 Top Countries With Highest Total Tests.**

In [120]:
casesbar('total_tests')

**10.8 Top Countries With Highest Total Tests per 1m population.**

In [121]:
casesbar('total_tests_per_1m_population')

**11. CONTINENTS COVID 19 CASES**

In [122]:
total_cases_by_continents = summary_df.groupby('continent')[
    'total_confirmed'
].sum().sort_values(ascending = False)
total_cases_by_continents

,total_confirmed
continent,
Europe,194330079
Asia,149999659
North America,99625662
South America,57136485
Africa,12042400
Australia/Oceania,7942867


**12. COUNTRIES GRAPHS.**

In [123]:
def country_graphs(country):
    if country in ['USA', 'UK']:
        country_prefix = 'the'
    else:
        country_prefix = ''


    country_df = daily_df[daily_df.country == country]
    country_df.set_index(
        'date',
        inplace = True
    )

    #Cumulative Total Cases Plot
    if not all(country_df.cumulative_total_cases.isna()):
        layout = go.Layout(
            yaxis = {
                'range': [
                    0,
                    country_df.cumulative_total_cases[-1] * 1.05
                ],
                'title' : 'Coronavirus Confirmed Cases'
            },
            xaxis = {
                'title' : ''
            }
        )

        fig = px.area(
            country_df,
            x = country_df.index,
            y = 'cumulative_total_cases',
            title = f"<b>Cumulative Total Confirmed Cases in {country_prefix} {country}.</b>",
            template = 'plotly_dark'
        )

        fig.update_traces(
            line = {
                'width' :5
            }
        )
        fig.update_layout(layout)
        fig.show()


    #Daily New Cases Plot
    if not all(country_df.daily_new_cases.isna()):
        layout = go.Layout(
           yaxis = {
               'range':[
                   0, country_df.daily_new_cases.max() * 1.05
               ],
                'title' : 'Daily New Coronavirus Confirmed Cases'
            },
            xaxis = {'title':''},
            template = 'plotly_dark',
            title = f"<b>Daily New Cases in  {country_prefix} {country}.</b>"
        )

        moving_avg7 = country_df.daily_new_cases.rolling(7).mean().dropna().astype(int)

        fig = go.Figure()
        fig.add_trace(
            go.Bar(
                name = 'Daily Cases',
                x = country_df.index,
                y = country_df.daily_new_cases
            )
        )

        fig.add_trace(
            go.Scatter(
                name = "7 Day Moving Average",
                x = country_df.index[country_df.shape[0] - moving_avg7.shape[0]:],
                y = moving_avg7,
                line = dict(
                    width = 3
                )
            )
        )

        fig.update_layout(layout)
        fig.show()


    #Cumulative Total Deaths Plot
    if not all(country_df.cumulative_total_deaths.isna()):
        layout = go.Layout(
            yaxis = {
                'range' : [
                    0,
                    country_df.cumulative_total_deaths[-1] * 1.05
                ],
                'title':'Coronavirus Deaths'
            },
            xaxis = {
                'title':''
            }
        )

        fig = px.area(
            country_df,
            x = country_df.index,
            y = 'cumulative_total_deaths',
            title = f"<b>Cumulative Total Deaths in {country_prefix} {country}.</b>",
            template = 'plotly_dark'
        )

        fig.update_traces(
            line = {
                'color':'red',
                'width':5
            }
        )

        fig.update_layout(layout)
        fig.show()


    #Daily New Deaths PLot
    if not all(country_df.daily_new_deaths.isna()):
        layout = go.Layout(
            yaxis = {
                'range' : [
                    0,
                    country_df.daily_new_deaths.max() * 1.05
                ],
                'title': 'Daily New Coronavirus Deaths'
            },
            xaxis = {
                'title': ''
            },
            template = 'plotly_dark',
            title = f"<b>Daily Deaths in {country_prefix} {country}.</b>"
        )

        moving_avg7 = country_df.daily_new_deaths.rolling(7).mean().dropna().astype(int)

        fig = go.Figure()

        fig.add_trace(
            go.Bar(
                name = "Daily Deaths",
                x = country_df.index,
                y = country_df.daily_new_deaths,
                marker_color = 'red'
            )
        )

        fig.add_trace(
            go.Scatter(
                name = "7 Day Moving Average",
                x = country_df.index[country_df.shape[0] - moving_avg7.shape[0]:],
                y = moving_avg7,
                line = {
                    'width' : 3,
                    'color' : 'white'
                }
            )
        )

        fig.update_layout(layout)
        fig.show()


    #Active Cases
    if not all(country_df.active_cases.isna()):
        layout = go.Layout(
            yaxis = {
                'range' :[
                    0,
                    country_df.active_cases.max() * 1.05
                ],
                'title' : 'Active Coronavirus Cases'
            },
            xaxis = {
                'title' : ''
            }
        )

        fig = px.line(
            country_df,
            x = country_df.index,
            y = "active_cases",
            title = f"<b>Active Cases in {country_prefix} {country}.</b>",
            template = 'plotly_dark'
        )

        fig.update_traces (
            line = {
                'color': 'yellow',
                'width' :5
            }
        )

        fig.update_layout(layout)
        fig.show()

**12.1 USA.**

In [124]:
country_graphs('USA')

/tmp/ipython-input-840992416.py:20: FutureWarning:

Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`



/tmp/ipython-input-840992416.py:92: FutureWarning:

Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`



In [125]:
usa = summary_df[summary_df['country'] == 'USA']
usa

,country,continent,total_confirmed,total_deaths,total_recovered,active_cases,serious_or_critical,total_cases_per_1m_population,total_deaths_per_1m_population,total_tests,total_tests_per_1m_population,population,mortality_rate,fatality_rate
216,USA,North America,84209473,1026646.0,81244260.0,1938567.0,1941.0,251659,3068.0,1.016883e+09,3038939.0,334617623,0.306812,1.219157


**12.1 NIGERIA.**

In [126]:
country_graphs('Nigeria')

/tmp/ipython-input-840992416.py:20: FutureWarning:

Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`



/tmp/ipython-input-840992416.py:92: FutureWarning:

Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`



In [127]:
nigeria = summary_df[summary_df['country'] == 'Nigeria']
nigeria

,country,continent,total_confirmed,total_deaths,total_recovered,active_cases,serious_or_critical,total_cases_per_1m_population,total_deaths_per_1m_population,total_tests,total_tests_per_1m_population,population,mortality_rate,fatality_rate
149,Nigeria,Africa,255859,3143.0,249943.0,2773.0,11.0,1186,15.0,5114703.0,23717.0,215651424,0.001457,1.228411
